src/Microsoft.DotNet.Build.CloudTestTasks/SendJobsToHelix.cs

```diff
@@ -151,7 +157,19 @@ public override bool Execute()
                             Log.LogWarning("Exception thrown attempting to submit job to Helix:");
                             Log.LogWarningFromException(toLog, true);
                         }
-
+                        catch (TaskCanceledException possibleClientTimeout)
+                        {
+                            if (possibleClientTimeout.CancellationToken != cancelTokenSource.Token)
+                            {
+                                // This is a timeout.
+                                Log.LogWarning("HttpClient timeout while POSTing new job; will retry (if tries remaining).");
```

Comment:
We could choose instead to set keepTrying false here, but:
- In practice when this fails to post a job, I have yet to see the POST actually succeed behind the scenes
- In the rare chance where I'm wrong, posting the same job twice will only result in the work being done twice, which is usually preferable to crashing.